pkg/scheduler/backend/queue/scheduling_queue.go

```diff
@@ -599,7 +617,13 @@ func (p *PriorityQueue) Activate(logger klog.Logger, pods map[string]*v1.Pod) {
 	for _, pod := range pods {
 		if p.activate(logger, pod) {
 			activated = true
+			continue
 		}
+
+		// If this pod is in-flight, register the activation event so that the pod will be requeued when it comes back.
+		// Specifically in the in-tree plugins, this is for the scenario with the preemption plugin
+		// where the async preemption API calls are all done or fail at some point before the Pod comes back to the queue.
+		p.activeQ.addEventsIfPodInFlight(nil, pod, []framework.ClusterEvent{framework.EventForceActivate})
```

Comment:
What if queueing hints feature is disabled and we reach this path? As inFlightEvents are not used then, pod will stay in unschedulable queue?

Reply:
Good point! We have to update `moveRequestCycle` like we do (when QHint is disabled) every time receiving an event.